Dockerfile

```diff
@@ -1,31 +1,29 @@
-
 FROM osgeo/gdal:ubuntu-small-3.3.1
 
 ENV DEBIAN_FRONTEND=noninteractive \
     LC_ALL=C.UTF-8 \
-    LANG=C.UTF-8
+    LANG=C.UTF-8 \
+    CIABPASSWORD=secretpassword
 
 RUN apt-get update && \
     apt-get install -y \
       build-essential \
-      fish \
       git \
       # For Psycopg2
-      libpq-dev python-dev \
+      libpq-dev python3-dev \
       python3-pip \
       tini \
       wget \
-    && apt-get autoclean && \
-    apt-get autoremove && \
-    rm -rf /var/lib/{apt,dpkg,cache,log}
+    && apt-get autoclean \
+    && apt-get autoremove \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}
 
-RUN mkdir -p /conf
 COPY requirements.txt /conf/
-RUN pip install -r /conf/requirements.txt \
-    && rm -rf /var/lib/apt/lists/*
+RUN pip3 install --requirement /conf/requirements.txt \
+      && rm -rf /root/.cache
 
 WORKDIR /notebooks
 
 ENTRYPOINT ["/bin/tini", "--"]
 
-CMD ["jupyter", "notebook", "--allow-root", "--ip='0.0.0.0'" "--NotebookApp.token='secretpassword'"]
+CMD ["jupyter", "notebook", "--allow-root", "--ip='0.0.0.0'", "--NotebookApp.token=\"$CIABPASSWORD\""]
```

Comment:
Hmm, I see what you're doing. Let's make this a build argument.

It gets overwritten when you change the run command anyway.